internal/util/update/resource-merge.go

```diff
@@ -59,9 +60,15 @@ func (u ResourceMergeUpdater) Update(options UpdateOptions) error {
 	}
 	defer os.RemoveAll(updated.AbsPath())
 
-	err := setters.PerformAutoSetters(updated.AbsPath())
-	if err != nil {
-		return err
+	if options.AutoSet {
```

Comment:
We should probably add a test for each of the scenarios outlined in the design doc.

Reply:
Updated the logic. Assuming update logic works correctly and well tested, this change would only set auto-setters for the merged package if isSet flag is not true. 

All test cases related to updated can be found here https://github.com/kubernetes-sigs/kustomize/blob/master/kyaml/yaml/merge3/merge3_test.go